server/src/test/java/io/crate/integrationtests/SubSelectIntegrationTest.java

```diff
@@ -812,4 +813,49 @@ WHERE EXISTS (SELECT 1 FROM sys.cluster WHERE 1 = 2) order by height desc limit
         execute(stmt);
         assertThat(response.rowCount()).isEqualTo(15L);
     }
+
+    @Test
+    @UseRandomizedSchema(random = false)
+    public void test_cte_left_join() {
+
+        execute("CREATE TABLE foo (id TEXT, createdAt TIMESTAMP WITH TIME ZONE)");
+        execute("CREATE TABLE bar (id TEXT, createdAt TIMESTAMP WITH TIME ZONE)");
+
+        execute("INSERT INTO foo VALUES ('123', '2023-01-01'), ('456', '2023-02-01')");
+        execute("INSERT INTO bar VALUES ('abc', '2023-01-01'), ('def', '2023-01-01')");
+
```

Comment:
I think this is node needed - can replace it by a making configureExecution visible for testing and create a dedicated unit test for it.